actors/market/src/lib.rs

```diff
@@ -210,15 +210,10 @@ impl Actor {
             ));
         }
 
-        let (_, worker, controllers) = request_miner_control_addrs(rt, provider_id)?;
         let caller = rt.message().caller();
-        let mut caller_ok = caller == worker;
-        for controller in controllers.iter() {
-            if caller_ok {
-                break;
-            }
-            caller_ok = caller == *controller;
-        }
+        let caller_ok =
+            MinerActor::is_controlling_address(rt, IsControllingAddressParam { address: caller })?
+                .is_controlling;
```

Comment:
You need to use a message send to communicate with the miner actor. Inspect the request_miner_control_addrs method that you should be replacing, and follow those patterns.

Reply:
done